tools/build.py

```diff
@@ -103,9 +102,20 @@ def configure(self: 'Routines') -> None:
         if not self._params.build_directory.is_dir():
             self._params.build_directory.mkdir()
 
-        use_presets = pathlib.Path.cwd().joinpath('CMakeUserPresets.json').is_file()
         print('configuring for CMake build configs: ' + ', '.join(self._params.build_configs))
+        if self._params.is_generator_multiconfig:
+            command = [
+                'cmake',
+                '-G', self._params.generator,
+                '-B', str(self._params.build_directory),
+                '-S', str(pathlib.Path.cwd()),
+                self._decorate_cmake_variable('CMAKE_EXPORT_COMPILE_COMMANDS', 'ON', 'BOOL'),
+                self._decorate_cmake_variable('CMAKE_CONFIGURATION_TYPES', ';'.join(self._params.build_configs))
```

Comment:
I would not set this variable explicitly if the user has not explicitly set configuration types on the command line.